src/main/java/jcifs/smb1/netbios/Lmhosts.java

```diff
@@ -0,0 +1,158 @@
+/* jcifs smb client library in Java
+ * Copyright (C) 2000  "Michael B. Allen" <jcifs at samba dot org>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs.smb1.netbios;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Hashtable;
+
+import jcifs.smb1.Config;
+import jcifs.smb1.smb1.SmbFileInputStream;
+import jcifs.smb1.util.LogStream;
+
+import java.net.UnknownHostException;
+
+public class Lmhosts {
+
+    private static final String FILENAME = Config.getProperty( "jcifs.smb1.netbios.lmhosts" );
+    private static final Hashtable TAB = new Hashtable();
+    private static long lastModified = 1L;
+    private static int alt;
+    private static LogStream log = LogStream.getInstance();
+
+    /**
+     * This is really just for {@link jcifs.smb1.UniAddress}. It does
+     * not throw an {@link java.net.UnknownHostException} because this
+     * is queried frequently and exceptions would be rather costly to
+     * throw on a regular basis here.
+     */
+
+    public synchronized static NbtAddress getByName( String host ) {
+        return getByName( new Name( host, 0x20, null ));
+    }
+
+    synchronized static NbtAddress getByName( Name name ) {
+        NbtAddress result = null;
+
+        try {
+            if( FILENAME != null ) {
+                File f = new File( FILENAME );
+                long lm;
+
+                if(( lm = f.lastModified() ) > lastModified ) {
+                    lastModified = lm;
+                    TAB.clear();
+                    alt = 0;
+                    populate( new FileReader( f ));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileReader*. Currently, there are execution paths that do not contain closure statements, for example, when Lmhosts.populate throws an exception. Either a) close the object returned by *FileReader()* in a try-finally block or b) close the resource by declaring the object returned by *FileReader()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).